### PR TITLE
Make Kafka Consumer Group Configurable

### PR DESCRIPTION
**Pull request by ahmeroxa**
### Description

This PR introduces a new config option `groupId` that allows the end user to configure the Kafka Consumer Group ID to be used with the Source connector.

This will enable some valuable use cases such as scaling via multiple Conduit instances as well as recovery (where I can create a new instance and have it continue where the previous consumer ended).

Fixes # https://github.com/ConduitIO/conduit/issues/142

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-kafka/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Comment by ahmeroxa** [comment]
@hariso here is the PR. Would appreciate some help with adding tests for the change though.

File: README.md [hunk 1]
```diff
@@ -52,3 +52,4 @@ There's no global, connector configuration. Each connector instance is configure
 | `acks`               | destination         | The number of acknowledgments required before considering a record written to Kafka. Valid values: none, one, all.                                                                 | false    | `all`         |
 | `deliveryTimeout`    | destination         | Message delivery timeout.                                                                                                                                                          | false    | `10s`         |
 | `readFromBeginning`  | source              | Whether or not to read a topic from beginning (i.e. existing messages or only new messages).                                                                                       | false    | `false`       |
+| `groupId`            | source              | The Consumer Group ID to use for the Kafka Consumer on the Source connector.                                                                                                       | false    |               |
```
Comment (hariso) [inline README.md:55]: Nitpick: maybe groupID?

**Review by hariso — APPROVED** [review]
:rocket:

File: README.md (hunk 1 above)
Comment (ahmeroxa) [reply README.md:55]: That is the Go way 👍

Comment (ahmeroxa) [reply README.md]: Done

**Pull request merged**